src/react-apps/ux-editor/src/sagas/api/apiSagas.ts

```diff
@@ -191,14 +191,23 @@ function* apiFetchList(connectionDef: any, externalApisById: any, components: an
   }
 }
 
-function* apiCheckValue(connectionDef: any, lastUpdatedDataBinding: any, lastUpdatedDataValue: any,
-  formData: any, externalApisById: any, components: any, model: any) {
+function* apiCheckValue(connectionDef: any, lastUpdatedDataBinding: IDataModelFieldElement, lastUpdatedDataValue: any,
+  formData: any, externalApisById: any, components: IFormDesignerComponent, model: any, dataModelGroup?: string, index?: number) {
   for (const param in connectionDef.clientParams) {
     if (!param) {
       continue;
     }
-    if (!formData[connectionDef.clientParams[param]]) {
-      break;
+
+    const isPartOfRepeatingGroup: boolean = (dataModelGroup !== null && index !== null);
```

Comment:
I think we should consider adding `repeating` parameter to this check, in case data model group is set even if the group is not repeating. Or, update the call from container and do the check there, so that one of these is `null` if the container is not repeating. Then you avoid having to change the method signatures again :) 

Tip: no need to actually check if something is not null, you can just check if it _is_:
```suggestion
    const isPartOfRepeatingGroup: boolean = (dataModelGroup && index);
```

Reply:
Ah, good catch. I'll update the methods to include a repeating param.